hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1977,21 +1979,27 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
 	configs = append(configs, nodeTuningGeneratedConfigs.Items...)
 
 	for _, config := range configs {
-		manifestRaw := config.Data[TokenSecretConfigKey]
-		manifest, err := defaultAndValidateConfigManifest([]byte(manifestRaw))
-		if err != nil {
-			errors = append(errors, fmt.Errorf("configmap %q failed validation: %w", config.Name, err))
-			continue
+		cmPayload := config.Data[TokenSecretConfigKey]
+		// ignition config-map payload may have multiple yaml documents
+		manifestList := strings.Split(cmPayload, yamlDocDelimiter)
+		for _, manifestRaw := range manifestList {
+			if strings.TrimSpace(manifestRaw) == "" {
```

Comment:
could this trim(or something else) result in a slightly different manifestRaw, and so a different allConfigPlainText for existing Configs? hence resulting in a different configHash, hence triggering a rolling upgrade for existing clusters with conflicting yaml in the MC secret?

Reply:
Wondering if the (de)serializers can be configured to consume multiple documents in one file? Would want to cross-reference e.g. `kubectl create -f` to see how it's handled